world-chain-builder/src/payload/builder.rs

```diff
@@ -406,8 +408,17 @@ where
     }
 
     if !attributes.no_tx_pool {
+        let mut invalid_txs = vec![];
         let verified_gas_limit = (verified_blockspace_capacity as u64 * block_gas_limit) / 100;
         while let Some(pool_tx) = best_txs.next() {
+            if let Some(conditional_options) = pool_tx.transaction.conditional_options() {
```

Comment:
I think this should be fine, but instead of computing the state on `BlockId::latest()` we should be computing it on `BlockId::pending()`